posix.mak

```diff
@@ -39,6 +39,10 @@ install: all
 	mkdir -p $(INSTALL_DIR)/man
 	cp -r docs/man/* $(INSTALL_DIR)/man/
 
+check-clean-git:
+	@git diff-index --quiet HEAD -- || \
+		(echo "ERROR: You created temporary files. Store them in tests_results or remove them."; git status; exit 1)
```

Comment:
The error message sounds a bit too much like it is accusing the current user, which may not be at fault for the temporary files.  How about "ERROR:  Found residual temporary files.  Temporary files should be stored in `test_results` or explicitly removed."  

Also, the `git status` tends to overwhelm the error message, making it less obvious.  Perhaps use `git status -s` which displays a less verbose output.  In that case you may want to change the error message to "ERROR:  Found the following residual temporary files.  Temporary files should be stored in `test_results` or explicitly removed."

Example output:
```
ERROR:  Found the following residual temporary files.  Temporary files should be stored in `test_results` or explicitly removed.
?? dub.selections.json
```

Sorry for being so picky about such things.  I can't help it.

Reply:
Also, it might be helpful to add a comment with a link to this PR.  It helps when maintaining these makefiles to understand why certain things exist.

Reply:
Also, it might be helpful to add a comment with a link to this PR. It helps when maintaining these makefiles to understand why certain things exist.